Assets/Scripts/PlayerCharacter/PlayerController.cs

```diff
@@ -70,42 +66,50 @@ private void ProcessMovement()
             rb.AddForce(velocityDelta, ForceMode.VelocityChange);
         }
 
-        private void ProcessRotation()
+        private Vector3 CalculateVelocityDelta()
         {
-            if (inputMove.sqrMagnitude > 0.1f)
-            {
-                float targetAngle = Mathf.Atan2(inputMove.x, inputMove.z) * Mathf.Rad2Deg;
-                float newAngle = Mathf.SmoothDampAngle(transform.eulerAngles.y, targetAngle, ref rotationSpeed, 0.1f);
-                transform.rotation = Quaternion.Euler(0, newAngle, 0);
-            }
+            var cameraForward = cameraTransform.forward;
+            cameraForward.y = 0f;
+            var inputDirection = Quaternion.LookRotation(cameraForward) * inputMove;
+
+            var velocity = rb.velocity;
+            var targetVelocity = inputDirection * moveSpeed + new Vector3(0, velocity.y, 0);
+            return Vector3.MoveTowards(velocity, targetVelocity,
+                acceleration * Time.fixedDeltaTime) - rb.velocity;
         }
 
         private void GroundCheck()
```

Comment:
With some luck, player can jump twice resulting in twice as high jump. You can add `velocity.y < 0.01` as additional check to prevent this